java-client/src/main/java/org/opensearch/client/transport/httpclient5/ApacheHttpClient5TransportBuilder.java

```diff
@@ -263,6 +264,16 @@ public ApacheHttpClient5TransportBuilder setChunkedEnabled(boolean chunkedEnable
         return this;
     }
 
+    /**
+     * Optional custom tls strategy to replace the default
+     *
+     * @param tlsStrategy custom tlsStrategy
+     */
+    public ApacheHttpClient5TransportBuilder setTlsStrategy(TlsStrategy tlsStrategy) {
```

Comment:
I think providing the way to customize the TlsStrategy would be beneficial (right now you could do that by providing own `connectionManager` instance, which is verbose). However we should do that iin a way to keep API concise, relying on `XxxCallback` hooks, for example `RequestConfigCallback`, `HttpClientConfigCallback`.

```
    public interface ClientTlsStrategyConfigCallback {
        ClientTlsStrategyBuilder customizeClientTlsStrategyClientTlsStrategyBuilder clientTlsStrategyBuilder);
    }
```